mmf/trainers/core/training_loop.py

```diff
@@ -152,3 +143,16 @@ def _extract_loss(self, report: Dict[str, Any]) -> Tensor:
         loss_dict = report.losses
         loss = sum([loss.mean() for loss in loss_dict.values()])
         return loss
+
+    def _calculate_max_updates(self):
+        max_updates = self.training_config.max_updates
+        if max_updates is None:
+            max_updates = math.inf
+
+        max_epochs = self.training_config.max_epochs
+        if max_epochs is None:
+            epoch_max_updates = math.inf
+        else:
+            epoch_max_updates = len(self.train_loader) * max_epochs
+
+        return min(epoch_max_updates, max_updates)
```

Comment:
As said above, let's be mutually exclusive and avoid stuff that can be confusing to the end user.

Reply:
It would be great if you have time add some tests for this.

Reply:
Talked offline, decided not to make them mutually exclusive. Favor max_epochs over max_updates. Tests will be added in the following PR.